modules/core/src/main/java/org/apache/ignite/internal/cache/query/index/IndexQueryProcessor.java

```diff
@@ -125,6 +131,38 @@ public <K, V> IndexQueryResult<K, V> queryLocal(
 
                 IgniteBiTuple<K, V> row = currVal;
 
+                if (cctx.events().isRecordable(EVT_CACHE_QUERY_OBJECT_READ) &&
```

Comment:
Let's cache the result of `isRecordable` for the iterator.

Reply:
Boolean variable `isRecordable` is introduced in the `queryLocal()` method.